vcd/resource_vcd_nsxt_alb_virtual_service.go

```diff
@@ -0,0 +1,346 @@
+	}
+
+	updateVirtualServiceConfig, err := getNsxtAlbVirtualServiceType(d)
+	if err != nil {
+		return diag.Errorf("error getting NSX-T ALB Virtual Service type: %s", err)
+	}
+	updateVirtualServiceConfig.ID = d.Id()
+
+	_, err = albVirtualService.Update(updateVirtualServiceConfig)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error updating NSX-T ALB Virtual Service: %s", err))
+	}
+
+	return resourceVcdAlbVirtualServiceRead(ctx, d, meta)
+}
+
+func resourceVcdAlbVirtualServiceRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+
+	albVirtualService, err := vcdClient.GetAlbVirtualServiceById(d.Id())
+	if err != nil {
+		if govcd.ContainsNotFound(err) {
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(fmt.Errorf("could not retrieve NSX-T ALB Virtual Service: %s", err))
+	}
+
+	err = setNsxtAlbVirtualServiceData(d, albVirtualService.NsxtAlbVirtualService)
+	if err != nil {
+		return diag.Errorf("error setting NSX-T ALB Virtual Service data: %s", err)
+	}
+	d.SetId(albVirtualService.NsxtAlbVirtualService.ID)
+	return nil
+}
+
+func resourceVcdAlbVirtualServiceDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+	vcdClient.lockParentEdgeGtw(d)
+	defer vcdClient.unLockParentEdgeGtw(d)
+
+	albPool, err := vcdClient.GetAlbVirtualServiceById(d.Id())
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("could not retrieve NSX-T ALB Virtual Service: %s", err))
+	}
+
+	err = albPool.Delete()
+	if err != nil {
+		return diag.Errorf("error deleting NSX-T ALB Virtual Service: %s", err)
+	}
+
+	return nil
+}
+
+func resourceVcdAlbVirtualServiceImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	log.Printf("[TRACE] NSX-T ALB Virtual Service import initiated")
+
+	resourceURI := strings.Split(d.Id(), ImportSeparator)
+	if len(resourceURI) != 4 {
+		return nil, fmt.Errorf("resource name must be specified as org-name.vdc-name.nsxt-edge-gw-name.virtual_service_name")
+	}
+	orgName, vdcName, edgeName, virtualServiceName := resourceURI[0], resourceURI[1], resourceURI[2], resourceURI[3]
+
+	vcdClient := meta.(*VCDClient)
+
+	_, vdc, err := vcdClient.GetOrgAndVdc(orgName, vdcName)
+	if err != nil {
+		return nil, fmt.Errorf("unable to find Org %s: %s", vdcName, err)
+	}
+
+	if vdc.IsNsxv() {
+		return nil, fmt.Errorf("ALB Virtual Services are only supported on NSX-T. Please use 'vcd_lb_virtual_server' for NSX-V load balancers")
+	}
+
+	edge, err := vdc.GetNsxtEdgeGatewayByName(edgeName)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve NSX-T edge gateway with ID '%s': %s", d.Id(), err)
+	}
+
+	albVirtualService, err := vcdClient.GetAlbVirtualServiceByName(edge.EdgeGateway.ID, virtualServiceName)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve NSX-T ALB Virtual Service '%s': %s", virtualServiceName, err)
```

Comment:
There's some inconsistency in capitalization: in some places it's `Virtual Service`, while in others - `virtual service`.

Reply:
Yes. Used search now to hopefully catch all.